cmd/diff.go

```diff
@@ -195,18 +214,64 @@ func populateDiffRow(rootMetrics *profiler.CallMetrics, correlation *correlatedM
 
 	// Populate measurement columns
 	for profileIndex, metrics := range correlation.metrics {
-		baseIndex := profileIndex*len(diffCols) + 1
-		for dIndex, dType := range diffCols {
-			row[baseIndex+dIndex] = fmtDiff(rootMetrics, correlation.metrics[0], metrics, dType, threshold, df)
+		baseIndex := profileIndex*len(dp.columns) + 1
+		for dIndex, dType := range dp.columns {
+			row[baseIndex+dIndex] = dp.fmtDiff(rootMetrics, correlation.metrics[0], metrics, dType)
 		}
 	}
 	t.Append(row)
 }
 
+// detectTimeUnit iterates through the list of correlated metrics and tries to
+// figure out best displayUnit that can represent all displayable values.
+func (dp *diffPrinter) detectTimeUnit(correlations []*correlatedMetrics) displayUnit {
+	var val time.Duration
+	var unit displayUnit = displayUnitMs
+
+	for _, correlation := range correlations {
+		for _, metrics := range correlation.metrics {
+			if metrics == nil {
+				continue
+			}
+			for _, dType := range dp.columns {
+				switch dType {
+				case tableColTotal:
+					val = metrics.TotalTime
+				case tableColMin:
+					val = metrics.MinTime
+				case tableColMax:
+					val = metrics.MaxTime
+				case tableColMean:
+					val = metrics.MeanTime
+				case tableColMedian:
+					val = metrics.MedianTime
+				case tableColP50:
+					val = metrics.P50Time
+				case tableColP75:
+					val = metrics.P75Time
+				case tableColP90:
+					val = metrics.P90Time
+				case tableColP99:
+					val = metrics.P99Time
+				default:
+					continue
+				}
+
+				dUnit := detectTimeUnit(val)
+				if dUnit > unit {
```

Comment:
time units are sorted so ms > us > ns; this test checks whether we need more accuracy than what we have detected this far